File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans;
        int curr = 1;
        ans.push_back(1);
        for(int i = 1; i < nums.size(); i++) {
            ans.push_back(ans[i - 1] * nums[i - 1]);
        }

        for(int i = nums.size() - 1; i >= 0; i--) {
            ans[i] *= curr;
            curr *= nums[i];
        }

        return ans;
    }
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function (nums) {
  let curr = 1;
  const ans = [];
  ans.push(curr);

  for (let i = 1; i < nums.length; i++) {
    ans.push(ans[i - 1] * nums[i - 1]);
  }
  console.log(ans)

  curr = 1;

  for (let i = nums.length - 2; i >= 0; i--) {
    curr = curr * nums[i + 1];
    ans[i] = curr * ans[i];
  }

  return ans;
};